experiments/dorop-20/metadata/parsers.py

```diff
@@ -0,0 +1,201 @@
+            for filename in filenames:
+                full_path = dirpath / filename
+                paths.append(full_path)
+        return paths
+
+    def find_root_metadata_file_path(self) -> Path | None:
+        for file_path in self.get_file_paths(self.content_path):
+            if str(file_path).endswith(self.root_metadata_file_suffix):
+                return file_path
+        return None
+
+    def __init__(self, content_path: Path):
+        self.content_path: Path = content_path
+
+        file_path = self.find_root_metadata_file_path()
+        if not file_path:
+            raise MetadataFileNotFoundError
+        self.file_path: Path = file_path
+        self.root_tree = ElementAdapter.from_string(self.file_path.read_text(), self.namespaces)
+
+    def get_identifier(self) -> str:
+        return self.root_tree.get("OBJID")
+
+    def get_record_status(self) -> RecordStatus:
+        mets_header = self.root_tree.find("METS:metsHdr")
+        status = mets_header.get("RECORDSTATUS")
+        return RecordStatus[status.upper()]
+
+    def get_rights_info(self) -> str | None:
+        md_rights_elem = self.root_tree.find(".//METS:md[@USE='RIGHTS']")
+        child = md_rights_elem.get_children()[0]
+        if child.tag == "METS:mdRef":
+            return child.get("LOCREF")
+        else:
+            return None
+
+    def get_events(self) -> list[PreservationEvent]:
+        event_elems = self.root_tree.findall(".//PREMIS:event")
+        return [PremisEventParser(event_elem).get_event() for event_elem in event_elems]
+
+    def get_common_metadata(self) -> CommonMetadata:
+        md_elem = self.root_tree.find(".//METS:md[@USE='DESCRIPTIVE/COMMON']")
+        mdref_elem = md_elem.find("METS:mdRef")
+        metadata_rel_path = mdref_elem.get("LOCREF")
+        metadata_file_path = apply_relative_path(self.file_path.parent, Path(metadata_rel_path))
+        return CommonMetadataParser(metadata_file_path).get_metadata()
+
+    def get_repository_item(self) -> RepositoryItem:
+        struct_map_elem = self.root_tree.find(".//METS:structMap")
+        struct_map_id = struct_map_elem.get("ID")
+        struct_map_type = struct_map_elem.get("TYPE")
+        
+        order_elems = struct_map_elem.findall(".//METS:div[@ORDER]")
+
+        assets = []
+        struct_map_items = []
+        for order_elem in order_elems:
+            order_number = int(order_elem.get("ORDER"))
+            label = order_elem.get("LABEL")
+            mptr_elem = order_elem.find("METS:mptr")
+            asset_file_name: str = mptr_elem.get("LOCREF")
+            asset_file_path = apply_relative_path(self.file_path.parent, Path(asset_file_name))
+            asset = MetsAssetParser(asset_file_path).get_asset()
+            assets.append(asset)
+            struct_map_items.append(StructMapItem(
+                order=order_number,
+                label=label,
+                asset_id=asset.id
+            ))
```

Comment:
One thing I've opted not to think about yet is what happens if the asset files are not present (as might happen during an update). Ideally speaking we just allow missing documents, but right now I'm relying on them for getting the asset IDs for the struct map -- so we'd have to figure that out.